File: KolkoiKrzyzyk/static/checkWinner.js
function checkWinner(moves,winningLines) {

    for (const line of winningLines) {

        const values = line.map(key => moves.get(key));//makes table(n) of values

        const unique = new Set(values);//non unique are removed

        if (unique.size === 1 && !unique.has(undefined)) {
            return [...unique][0]; // "X" albo "O"
        }
    }
    return undefined; // brak zwycięzcy
}

File: KolkoiKrzyzykTinker/checkWinner.py
def checkWinner(moves,winningLines) :
    for  line in winningLines :

        values=[]
        values = [moves.get(i) for i in line]
        unique = set(values)
    # sprawdzenie warunku

        if len(unique) == 1 and None not in unique:  # Python używa None zamiast undefined
           result = next(iter(unique))  # pobiera jedyną wartość ze zbioru
           return result
        #print(result)  # np. "X" albo "O"
    return None# brak zwycięzcy
